File: C++/CheckTree.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
#include <iostream>
using namespace std;
struct TreeNode {
    int val;
    TreeNode *left;
    TreeNode *right;
    TreeNode() : val(0), left(nullptr), right(nullptr) {}//这个构造函数将节点的值初始化为 0，左子节点和右子节点都设置为 nullptr，即节点没有子节点。
    TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}//这个构造函数将节点的值初始化为 x，左子节点和右子节点都设置为 nullptr。
    TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}//这个构造函数除了初始化节点的值外，还初始化左右子节点。left 和 right 是指向二叉树左右子节点的指针。
};
class Solution {
public:
    bool checkTree(TreeNode* root) {
        return root->val == root->left->val + root->right->val;
    }
};

main(int argc, char const *argv[])
{
    Solution solution;
    TreeNode* root = new TreeNode(10);
    root->left = new TreeNode(4);
    root->right = new TreeNode(5);
    cout << solution.checkTree(root) << endl;
    return 0;
}

File: Python/CheckTree.py
# 定义二叉树节点类
class TreeNode:
    def __init__(self, val=0, left=None, right=None):
        self.val = val
        self.left = left
        self.right = right

# 定义Solution类，包含checkTree方法
class Solution:
    def checkTree(self, root: TreeNode) -> bool:
        # 检查根节点值是否等于左右子节点值之和
        return root.val == root.left.val + root.right.val

# 创建测试用例
if __name__ == "__main__":
    # 创建一个符合条件的二叉树：根节点值为3，左节点为1，右节点为2
    node1 = TreeNode(1)
    node2 = TreeNode(2)
    root = TreeNode(3, node1, node2)

    # 创建Solution对象
    solution = Solution()
    
    # 调用checkTree方法进行测试
    print(solution.checkTree(root))  # 应输出True

    # 创建一个不符合条件的二叉树：根节点值为4，左节点为1，右节点为2
    root2 = TreeNode(4, node1, node2)
    
    # 再次调用checkTree方法进行测试
    print(solution.checkTree(root2))  # 应输出False